File: app/src/main/java/com/example/asoro/healthygroceryassistant/db/RecipeWithIngredients.java
package com.example.asoro.healthygroceryassistant.db;

import java.util.List;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Relation;

import com.example.asoro.healthygroceryassistant.model.Ingredient;
import com.example.asoro.healthygroceryassistant.model.Recipe;

public class RecipeWithIngredients {
	@Embedded
	Recipe mRecipe;

	@Relation(parentColumn = "uri", entityColumn = "recipeUri", entity = Ingredient.class)
	List<Ingredient> mIngredients;

	public Recipe getRecipe() {
		return mRecipe;
	}

	public List<Ingredient> getIngredients() {
		return mIngredients;
	}
}

File: app/src/main/java/com/example/asoro/healthygroceryassistant/db/RecipeWithIngredients.kt
package com.example.asoro.healthygroceryassistant.db

import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Relation
import com.example.asoro.healthygroceryassistant.model.Ingredient
import com.example.asoro.healthygroceryassistant.model.Recipe


class RecipeWithIngredients {
    @Embedded
    var recipe: Recipe? = null

    @Relation(parentColumn = "recipe.uri", entityColumn = "recipeUri", entity = Ingredient::class)
    var ingredients:List<*>?=null
}